Failing test: Test_PullTemplates

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behaviour
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->

All unit tests should pass.

## Current Behaviour
<!--- If describing a bug, tell us what happens instead of the expected behaviour -->
<!--- If suggesting a change/improvement, explain the difference from current behaviour -->

The unit test `Test_PullTemplates` fails depending on the systems git configuration:

```
2023/09/22 16:33:04 No templates found in current directory.
2023/09/22 16:33:04 Attempting to expand templates from /tmp/openfaas-templates-test-1045834236
2023/09/22 16:33:04 Fetched 2 template(s) : [dockerfile ruby] from /tmp/openfaas-templates-test-1045834236
2023/09/22 16:33:04 Attempting to expand templates from /tmp/openfaas-templates-test-1045834236
Cloning into '/tmp/openfaas-templates-2918831397'...
warning: --depth is ignored in local clones; use file:// instead.
warning: Could not find remote branch master to clone.
fatal: Remote branch master not found in upstream origin
2023/09/22 16:33:04 Attempting to expand templates from /tmp/openfaas-templates-test-1045834236
2023/09/22 16:33:04 Fetched 2 template(s) : [dockerfile ruby] from /tmp/openfaas-templates-test-1045834236
--- FAIL: Test_PullTemplates (0.04s)
    --- FAIL: Test_PullTemplates/fetchTemplates_with_master_ref (0.01s)
        fetch_templates_test.go:30: exit status 128
        fetch_templates_test.go:78: Directory template was not created: stat ./template/: no such file or directory
    fetch_templates_test.go:78: Directory template was not created: stat ./template/: no such file or directory
FAIL
FAIL    github.com/openfaas/faas-cli/commands   0.039s
FAIL
```

<!-- How is this affecting you? What task are you trying to accomplish? -->
## Why do you need this?

<!-- Attempts to mask or hide this may result in the issue being closed -->
## Who is this for?

What company is this for? Are you listed in the [ADOPTERS.md](https://github.com/openfaas/faas/blob/master/ADOPTERS.md) file?

## Are you a GitHub Sponsor (Yes/No?)

<!-- Issues created by customers or monthly sponsors get priority -->

Check at: https://github.com/sponsors/openfaas
- [ ] Yes
- [x] No

## List All Possible Solutions and Workarounds
<!--- Suggest a fix/reason for the bug, or ideas how to implement  -->
<!--- the addition or change -->
<!--- Is there a workaround which could avoid making changes? -->

Ensure `master` is used as the default branch when setting up local repos for testing.

This can be done by adding the flag `--initial-branch=master` to the git init command.

## Which Solution Do You Recommend?
<!--- Pick your preferred solution, if you were to implement and maintain this change -->

Update the [GitInitRepo command](https://github.com/openfaas/faas-cli/blob/ab7202172f25f8bdb0a7f252c78f14afb31c9a41/versioncontrol/git.go#L41C1-L54) to use the suggested `--initial-branch` flag.

## Steps to Reproduce (for bugs)
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
1. Configure git to use main as the default init branch
    ```bash
    git config --global init.defaultBranch main
    ```
2. Run `go test -timeout 30s -run ^Test_PullTemplates$ github.com/openfaas/faas-cli/commands`

## Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* FaaS-CLI version ( Full output from: `faas-cli version` ):

```bash
CLI:
 commit:  d12888b56367056c23ae14082428bd53ba3dfcb0
 version: 0.16.14
```

* Docker version ( Full output from: `docker version` ):

* Are you using Docker Swarm (FaaS-swarm ) or Kubernetes (FaaS-netes)?

* Operating System and version (e.g. Linux, Windows, MacOS):

* Link to your project or a code example to reproduce issue:

--- a/versioncontrol/git.go
+++ b/versioncontrol/git.go
@@ -43,7 +43,7 @@ var GitInitRepo = &vcsCmd{
 	name: "Git",
 	cmd:  "git",
 	cmds: []string{
-		"init {dir}",
+		"init {dir} --initial-branch=master",
 		"config core.autocrlf false",
 		"config user.email \"contact@openfaas.com\"",
 		"config user.name \"OpenFaaS\"",

